File: packet-traces/output_to_csv.py
import re
import csv

# Regex to parse tcpdump line
line_re = re.compile(
    r'(?P<timestamp>\d+:\d+:\d+\.\d+)\s+IP\s+'
    r'(?P<src_ip>\d+\.\d+\.\d+\.\d+)\.(?P<src_port>\d+)\s+>\s+'
    r'(?P<dst_ip>\d+\.\d+\.\d+\.\d+)\.(?P<dst_port>\d+):.*length\s+(?P<length>\d+)'
)

def parse_tcpdump(input_file, output_file):
    with open(input_file, 'r') as infile, open(output_file, 'w', newline='') as outfile:
        writer = csv.writer(outfile)
        # Write CSV header
        writer.writerow(["timestamp", "source_ip", "source_port", "destination_ip", "destination_port", "packet_size"])

        for line in infile:
            match = line_re.search(line)
            if match:
                writer.writerow([
                    match.group("timestamp"),
                    match.group("src_ip"),
                    match.group("src_port"),
                    match.group("dst_ip"),
                    match.group("dst_port"),
                    match.group("length")
                ])

if __name__ == "__main__":
    parse_tcpdump("packets.log", "output.csv")

File: packet-traces/output_to_csv.rb
require 'csv'

# Regex to parse tcpdump line
LINE_RE = /
  (?<timestamp>\d+:\d+:\d+\.\d+)\s+IP\s+
  (?<src_ip>\d+\.\d+\.\d+\.\d+)\.(?<src_port>\d+)\s+>\s+
  (?<dst_ip>\d+\.\d+\.\d+\.\d+)\.(?<dst_port>\d+):.*length\s+(?<length>\d+)
/x

def parse_tcpdump(input_file, output_file)
  CSV.open(output_file, "w") do |csv|
    # Write CSV header
    csv << ["timestamp", "source_ip", "source_port", "destination_ip", "destination_port", "packet_size"]

    File.foreach(input_file) do |line|
      if (match = LINE_RE.match(line))
        csv << [
          match[:timestamp],
          match[:src_ip],
          match[:src_port],
          match[:dst_ip],
          match[:dst_port],
          match[:length]
        ]
      end
    end
  end
end

if __FILE__ == $0
  parse_tcpdump("packets.log", "output.csv")
end
